src/BinaryParsers/PEBinary/PortableExecutable/PE.cs

```diff
@@ -867,5 +867,43 @@ private ChecksumAlgorithmType ChecksumAlgorithmForFullPdb(Pdb pdb)
 
             return ChecksumAlgorithmType.Unknown;
         }
+        public bool IsMostlyOptimized(Pdb pdb)
+        {
+            uint count = 0;
+            uint optimizedCount = 0;
+            foreach (DisposableEnumerableView<Symbol> omView in pdb.CreateObjectModuleIterator())
+            {
+                Symbol om = omView.Value;
+                ++count;
+                if (om.GetObjectModuleDetails().OptimizationsEnabled)
+                {
+                    ++optimizedCount;
+                }
+            }
+
+            // Arbitrary threshold of 50% optimized.
+            // Things linked into the binary also count as symbols so this can be complicated.  There is probably a better heuristic than this.
+            // A small single-file win32 console application is just over 50% optimized.  A large project is more like 95%.
+            if (((float)optimizedCount / (float)count) > 0.50f)

```

Comment:
This heuristic is not ideal.  I'm looking for better options.

Reply:
Detection just from a binary? Tough one.

Imports to debug CRTs imply not-optimized

Similarly, if the binary has certain symbols from a statically linked debug CRT it's likely not-optimized. Not sure what symbols to look for. In particular in cases like WinAppSDK using the HybridCRT approach the Standard C library is UCRT and just the Standard C++ stuff (vector etc) are static. Not sure if there's any debug artifacts in there one could sniff out

Reply:
That's an interesting suggestion.  Linking to the debug C runtime definitely means a debug build.  However many projects link to the release C runtime, even for debug builds, so it is not a firm guarantee.  That might be a good early-exit approach.  I'll think about this more.